convert2rhel/toolopts.py

```diff
@@ -515,70 +539,118 @@ def _process_cli_options(self):
             )
 
 
-def _log_command_used():
-    """We want to log the command used for convert2rhel to make it easier to know what command was used
-    when debugging the log files. Since we can't differentiate between the handlers we log to both stdout
-    and the logfile
-    """
-    command = " ".join(utils.hide_secrets(sys.argv))
-    loggerinst.info("convert2rhel command used:\n{0}".format(command))
-
-
-def options_from_config_files(cfg_path=None):
+def options_from_config_files(cfg_path):
     """Parse the convert2rhel.ini configuration file.
 
     This function will try to parse the convert2rhel.ini configuration file and
     return a dictionary containing the values found in the file.
 
     .. note::
-       This function will parse the configuration file following a specific
-       order, which is:
-       1) Path provided by the user in cfg_path (Highest priority).
-       2) ~/.convert2rhel.ini (The 2nd highest priority).
-       3) /etc/convert2rhel.ini (The lowest priority).
+        This function will parse the configuration file in the following way:
+
+        1) If the path provided by the user in cfg_path is set (Highest
+           priority), then we use only that.
+
+        Otherwise, if cfg_path is `None`, we proceed to check the following
+        paths:
+
+        2) ~/.convert2rhel.ini (The 2nd highest priority).
+        3) /etc/convert2rhel.ini (The lowest priority).
+
+        In any case, they are parsed in reversed order, meaning that we will
+        start with the lowest priority and go until the highest.
 
     :param cfg_path: Path of a custom configuration file
     :type cfg_path: str
 
     :return: Dict with the supported options alongside their values.
-    :rtype: dict[str, str | None]
+    :rtype: dict[str, str]
     """
-    headers = ["subscription_manager"]  # supported sections in config file
-    # Create dict with all supported options, all of them set to None
-    # needed for avoiding problems with files priority
-    # The name of supported option MUST correspond with the name in ToolOpts()
-    # Otherwise it won't be used
-    supported_opts = {"username": None, "password": None, "activation_key": None, "org": None}
+    # Paths for the configuration files. In case we have cfg_path defined
+    # (meaning that the user entered something through the `-c` option), we
+    # will use only that, as it has a higher priority over the rest
+    config_paths = [cfg_path] if cfg_path else CONFIG_PATHS
+    paths = [os.path.expanduser(path) for path in config_paths if os.path.exists(os.path.expanduser(path))]
+
+    if cfg_path and not paths:
+        raise FileNotFoundError("No such file or directory: %s" % ", ".join(paths))
```

Comment:
This would be problem on systems with python 2.7 - this exception is not defined. I would go with
`raise OSError(2, "No such file or directory: '%s'" % cfg_path)`

Reply:
good point! thanks for suggestion